lib/Doctrine/DBAL/Driver/AbstractDriverException.php

```diff
@@ -36,6 +36,42 @@ public function __construct($message, $sqlState = null, $errorCode = null)
         $this->sqlState  = $sqlState;
     }
 
+    /**
+     * @return AbstractDriverException
+     */
+    public static function noInsertId() : self
+    {
+        $message = 'No identity value was generated by the last statement.';
+
+        if (static::class === self::class) {
+            // WIP regarding exceptions, see:
+            // https://github.com/doctrine/dbal/pull/3335#discussion_r234381175
+            return new class($message) extends AbstractDriverException {
```

Comment:
I had to resort to this nasty code, as I can't use a factory method on an anonymous class.
Exceptions refactoring is really needed here!

Reply:
Note that Scrutinizer wrongly reports that `static::class === self::class` is always true, I'll file a bug with them.